include/boost/random/detail/uniform_int_float.hpp

```diff
@@ -60,7 +60,7 @@ class uniform_int_float
     result_type operator()()
     {
         base_result range = static_cast<base_result>((max)())+1;
-        return static_cast<result_type>(_rng() * range);
+        return static_cast<result_type>(static_cast<base_result>(_rng() * range));
```

Comment:
AMDG

On 03/14/2015 05:55 AM, jzmaddock wrote:
>
>> base_result_type should be hidden from doxygen (and I'd also prefer 
> not to rename it).
> 
> This one is trickier: base_result_type is used as the type used to seed 
> by a single value: and since it just gets forwarded to the underlying 
> generator, this is actually the correct type for a single valued seed.  
> So the type *is* used in the interface, and is named to match the other 
> public typedefs.  The alternative would be to make the seed type the 
> same as the result type and explicitly cast to the base_result_type 
> internally.  However, this has two key disadvantages - it's C++11 only, 
> and it's also rather inefficient (we construct a wider/multiprecision 
> type, just to narrow it again internally).  Oh, and it's illogical too ;)
> 

Okay.  I don't like it, but there's no good solution.

> BTW there are rather a lot of commits now, let me know if you want me to 
> condense down to a single commit.
> 

That would be helpful.  A few commits should be
separate, though.  For example, adding missing
#includes to random.hpp.

In Christ,
Steven Watanabe